### PR TITLE
sql: benchmark for function expression type checking

### PR DESCRIPTION
**Pull request by chengxiong-ruan**
Release note: None

### CODE REVIEW

File: pkg/bench/bench_test.go [hunk 1]
```diff
@@ -1420,3 +1428,81 @@ func BenchmarkNameResolution(b *testing.B) {
...
+		},
+	}
+
+	for _, tc := range testCases {
+		b.Run(tc.name, func(b *testing.B) {
+			expr, err := parser.ParseExpr(tc.exprStr)
+			require.NoError(b, err)
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				for j := 0; j < 1000; j++ {
```
Comment (ajwerner) [inline pkg/bench/bench_test.go:1500]: What is this `j` part?

Comment (chengxiong-ruan) [reply pkg/bench/bench_test.go:1500]: the [scripts/bench](https://github.com/cockroachdb/cockroach/blob/master/scripts/bench#L49-L49) script use `10` as benchmark count which seems low. I think I'm just lazy here and give it a `1000` run so the script can have some how reasonable result when running this benchmark.

Comment (chengxiong-ruan) [reply pkg/bench/bench_test.go:1512]: Ah, I see, let me remove that.

**Review by ajwerner — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/89438#-:-NDhcouyAlhPHSpT4Di2:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @chengxiong-ruan)

**Review by ajwerner — APPROVED** [review]
Can you add a test for comparing tuples? Likes `(1, 2, 3) > (1, 2, 4)`

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/89438#-:-NDhcrXF6qPVpvND_Gzd:bs92n7q)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @chengxiong-ruan)

**Comment by chengxiong-ruan** [comment]
TFTR!
bors r+

**Pull request merged**